crates/sovereign-sdk/module-system/sov-modules-stf-blueprint/src/lib.rs

```diff
@@ -526,27 +523,16 @@ where
             }
         }
 
-        let mut filtered: Vec<SequencerCommitment> = vec![];
-        let mut visited_l2_ranges: HashSet<(u32, u32), RandomState> =
-            HashSet::from_iter(preproven_commitment_ranges);
-        for sequencer_commitment in sequencer_commitments {
-            // Handle commitments which have the same L2 range
-            let current_range = (
-                sequencer_commitment.l2_start_block_number as u32,
-                sequencer_commitment.l2_end_block_number as u32,
-            );
-            if visited_l2_ranges.contains(&current_range) {
-                continue;
-            }
-            visited_l2_ranges.insert(current_range);
-
-            filtered.push(sequencer_commitment.clone());
-        }
-
-        let mut sequencer_commitments = filtered;
         // Sort commitments just in case
         sequencer_commitments.sort_unstable();
 
+        let sequencer_commitments: Vec<_> = sequencer_commitments
+            .into_iter()
+            .enumerate()
+            .filter(|(index, _)| !preproven_commitment_indicies.contains(index))
```

Comment:
I guess we can use Iter::any instead of contains so that we don't rescan the preproven indices from the start. might be wrong on the function name but I'm %100 sure there is something like that. 

we'd just need to make sure the list is ever increasing.

Reply:
well couldn't really find what I was looking for, maybe it was an iter trick for some other condition..